Sources/MlemMiddleware/FeedLoaders/User Content Feed Loader/UserContentFeedLoader.swift

```diff
@@ -0,0 +1,240 @@
+//
+//  CompositeFeedLoader.swift
+//
+//
+//  Created by Eric Andrews on 2024-07-09.
+//
+
+import Foundation
+import Semaphore
+import Nuke
+
+/// This is a special type of FeedLoader built for user content, which is uniquely challenging because you cannot load
+/// just posts or just comments, and thus the standard Parent/Child FeedLoader construction does not work without
+/// severe API waste. This solution is a simplified variant of that architecture.
+///
+/// The UserContentFeedLoader is the parent loader. It is responsible for all data fetching, and keeps track of two
+/// UserContentStreams, one for Posts and one for Comments. To load a page of items, it consumes and merges the child streams, just as
+/// in the standard Parent/Child FeedLoader; if either stream reaches the end of its items, it triggers a new load, the response from
+/// which is then incorporated into both child streams.
+
+@Observable
+public class UserContentFeedLoader: FeedLoading {
+    public var api: ApiClient
+    public var items: [UserContent]
+    public var loadingState: LoadingState
+    
+    // loading configuration
+    private(set) var sortType: FeedLoaderSort.SortType
```

Comment:
Will this need to be read from Mlem?

```suggestion
    public private(set) var sortType: FeedLoaderSort.SortType
```